overlord/snapstate/snapmgr.go

```diff
@@ -209,6 +209,22 @@ func (snapst *SnapState) Block() []snap.Revision {
 
 var ErrNoCurrent = errors.New("snap has no current revision")
 
+// Retrieval functions
+var readInfo = readInfoImpl
+
+func readInfoImpl(name string, si *snap.SideInfo) (*snap.Info, error) {
+	info, err := snap.ReadInfo(name, si)
+	if _, ok := err.(*snap.NotFoundError); ok {
```

Comment:
Do we care that it was a not found error here?  Should we mention it is broken if something else goes wrong as well?